src/liballoc/rc.rs

```diff
@@ -900,12 +901,47 @@ impl<T: Default> Default for Rc<T> {
     }
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
+trait RcEqIdent<T: ?Sized + PartialEq> {
+    fn eq(&self, other: &Rc<T>) -> bool;
+    fn ne(&self, other: &Rc<T>) -> bool;
```

Comment:
Couldn't this also have an impl right here that just calls `!self.eq(other)`? That way the impls don't need the trivial implementations given.

Reply:
yeah, the current impl is doing this as well (<https://github.com/rust-lang/rust/blob/77a6a61f066af3dd693d8527a8a1bf5a446d295c/src/liballoc/rc.rs#L936-L939>) i don't know why it's not just using the default impl of ne. So i copied that behaviour.

Reply:
yes, you'r right

Reply:
but what if `T` has a custom impl which isn't `!self.eq(other)`, especially for the default (non-`Eq`) case?

Reply:
Oh right. Performance is a point here. Indeed, leave it in

Reply:
> Any manual implementation of ne must respect the rule that eq is a strict inverse of ne; that is, !(a == b) if and only if a != b.
>
> -- https://doc.rust-lang.org/std/cmp/trait.PartialEq.html

> Actually there may be good reason for the impls. LLVM is a strange beast sometimes. I certainly wouldn't remove the impls without prior careful measurement.
>
> -- https://github.com/rust-lang/rust-clippy/issues/86

so the whole point of `PartialEq::ne` even being a thing is performance?